llvm/lib/CodeGen/TargetPassConfig.cpp

```diff
@@ -506,6 +506,9 @@ static void registerPartialPipelineCallback(PassInstrumentationCallbacks &PIC,
   unsigned StopBeforeInstanceNum = 0;
   unsigned StopAfterInstanceNum = 0;
 
+  bool IsStopBeforeMachinePass = false;
```

Comment:
this patch is doing too much at once, please split out the stop before/after to a separate change